src/panels/config/devices/device-detail/integration-elements/zha/ha-device-actions-zha.ts

```diff
@@ -119,9 +122,18 @@ export class HaDeviceActionsZha extends LitElement {
       return;
     }
 
-    this.hass.callService("zha", "remove", {
-      ieee_address: this._zhaDevice!.ieee,
-    });
+    this.hass
+      .callService("zha", "remove", {
+        ieee_address: this._zhaDevice!.ieee,
+      })
+      .then(() => {
+        navigate(
+          this,
+          "/config/devices/dashboard?historyBack=1&config_entry=" +
+            this.configEntry.entry_id,
+          true
+        );
+      });
```

Comment:
```suggestion
    await this.hass
      .callService("zha", "remove", {
        ieee_address: this._zhaDevice!.ieee,
      });
    navigate(
      this,
      "/config/devices/dashboard?historyBack=1&config_entry=" +
        this.configEntry.entry_id,
      true
    );
```

Reply:
Can't we just navigate back?
`history.back();`

Reply:
yep. Didn't think of that :(

Reply:
fixed here: #6638